app/components/export-tool.js

```diff
@@ -4,72 +4,81 @@ function rawValue(value) {
     return value;
 }
 
+function flatten(obj) {
+    var ret = [];
+    Ember.$.each(Object.keys(obj), (_, key) => {
+        ret.push([key, obj[key]]);
+    });
+    return ret;
+}
+
+function squash(obj, prefix) {
+    prefix = prefix || '';
+
+    var ret = {};
+    if (obj.serialize) {
+        var serialized = obj.serialize();
+        obj = serialized.data.attributes;
+    }
+    Ember.$.each(Object.keys(obj), (_, key) => {
+        var value = Ember.get(obj, key);
+        if (value && value.toJSON) {
+            value = value.toJSON();
+        }
+
+        if (Ember.$.isPlainObject(value)) {
+            ret = Ember.$.extend({}, ret, squash(value, prefix ? `${prefix}.${key}` : key));
+        }
+        else {
+            ret[prefix ? `${prefix}.${key}` : key] = value;
+        }
+    });
+    return ret;
+}
+
 export default Ember.Component.extend({
     attributeBindings: ['data', 'mappingFunction'],
     dataFormat: 'JSON',
     dataFormats: [
         'JSON',
-        'CSV'
+        'TSV'
```

Comment:
If the item you're serializing is an array... it *can* be serialized in a CSV, but it's a bit trickier.  When the delimiter is a tab, it's less likely (though by no means impossible) to hit the annoying edge cases.

The conventions for CSV values with commas are:
1. Wrap the value of each field in quotation marks (`"bob", "[1,2,3]"` rather than `bob, [1,2,3]`)
2. Escape any existing quotation marks as appropriate